tests/jsunit/event_test.js

```diff
@@ -790,3 +790,30 @@ function test_events_newblock_newvar_xml() {
     Blockly.Events.fire = savedFireFunc;
   }
 }
+
+function test_events_filter_nomerge_move() {
+  // Move events should only merge if they refer to the same block and are
+  // consecutive.
+  eventTest_setUpWithMockBlocks();
+  try {
+    var block1 = createSimpleTestBlock(workspace);
+    var block2 = createSimpleTestBlock(workspace);
+
+    var events = [];
+    helper_addMoveEvent(events, block1, 1, 1);
```

Comment:
Can you add a comment about the shadow block case this attempts to emulate, giving the test some context?